### ISSUE TITLE
Add patch on Windows

### ISSUE THREAD

**Issue by smorimoto** [actions/virtual-environments#1552]
**Tool information**

- Tool name: GNU patch
- Tool license: GPL3
- Add or update? Add
- Desired version: latest
- Approximate size:
- If this is an add request:
  - Brief description of tool: Patch takes a patch file containing a difference listing produced by the diff program and applies those differences to one or more original files, producing patched versions.
  - URL for tool's homepage: https://savannah.gnu.org/projects/patch/
  - Provide a basic test case to validate the tool's functionality: `patch --version`

**Area for Triage**:

Packages

**Question, Bug, or Feature?**:

Feature

**Virtual environments affected**

- [ ] macOS 10.15
- [ ] Ubuntu 16.04 LTS
- [ ] Ubuntu 18.04 LTS
- [ ] Ubuntu 20.04 LTS
- [x] Windows Server 2016 R2
- [x] Windows Server 2019

**Can this tool be installed during the build?**

**Tool installation time in runtime**

**Are you willing to submit a PR?**

**Comment by miketimofeev** [comment]
Hi @smorimoto!
Is it possible to use `patch` from msys2 for your needs?
```
$origPath = $env:PATH
$env:PATH = "C:\msys64\mingw64\bin;C:\msys64\usr\bin;$origPath"
patch --version
```

**Comment by smorimoto** [comment]
In other words, does executing this command in the Actions VM make it work?

```cmd
ECHO "::add-path::C:\msys64\mingw64\bin"
```

**Comment by smorimoto** [comment]
I knew MSYS2 was pre-installed, but I didn't know that patch would be available by doing so. Great.

**Comment by miketimofeev** [comment]
@smorimoto yeah, basically you just need to msys64 directories to the PATH, `patch` lives in `C:\msys64\usr\bin`. Without adding to the PATH:
```
CommandType     Name                                               Version    Source                                   
-----------     ----                                               -------    ------                                   
Application     patch.exe                                          0.0.0.0    C:\Strawberry\c\bin\patch.exe            
patch 2.5.9
```
With msys2 in the PATH:
```
CommandType     Name                                               Version    Source                                   
-----------     ----                                               -------    ------                                           
Application     patch.exe                                          0.0.0.0    C:\msys64\usr\bin\patch.exe              
GNU patch 2.7.6
```

**Comment by CyberShadow** [comment]
> Is it possible to use `patch` from msys2 for your needs? MSYS2 is pre-installed on the image but not added to PATH.

This doesn't seem to be true any more. There is no `patch.exe` at those locations on `windows-2022`.

```
'"C:\msys64\usr\bin\patch.exe"' is not recognized as an internal or external command, operable program or batch file.
'"C:\msys64\mingw64\bin\patch.exe"' is not recognized as an internal or external command, operable program or batch file.
```

Git's `patch.exe` works, though. https://github.com/actions/runner-images/issues/5459#issuecomment-1532856844
